File: product/src/main/java/com/assoc/product/entity/Product.java
package com.assoc.product.entity;

import com.assoc.common.entity.AuditableEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "prd_product")
@Getter
@Setter
public class Product extends AuditableEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private ProductCategory category;

    private String manufacturer;

    private String model;

    private String price;

    @Column(columnDefinition = "TEXT")
    private String summary;

    @Column(columnDefinition = "TEXT")
    private String description;

    // JSON string fields
    @Column(columnDefinition = "TEXT")
    private String features;

    @Column(columnDefinition = "TEXT")
    private String application;

    @Column(columnDefinition = "TEXT")
    private String certifications;

    @Column(name = "contact_phone")
    private String contactPhone;

    @Column(name = "contact_email")
    private String contactEmail;

    private String contact;

    private String website;

    private Boolean featured = false;

    @Column(columnDefinition = "TEXT")
    private String images;

    @Column(columnDefinition = "TEXT")
    private String specifications;

    private Integer status = 0;

    private Long views = 0L;
}

File: web/src/types/product.ts
// 产品状态枚举（后端用数字）
export type ProductStatus = 0 | 1;  // 0=草稿, 1=已发布

// 状态标签映射
export const productStatusLabels: Record<ProductStatus, string> = {
  0: '草稿',
  1: '已发布',
};

// 产品分类响应
export interface ProductCategoryResponse {
  id: number;
  name: string;
  code: string;
  parentId: number | null;
  sortOrder: number;
  status: number;
  description: string | null;
  children?: ProductCategoryResponse[];
}

// 产品分类请求
export interface ProductCategoryRequest {
  name: string;
  code?: string;
  parentId?: number | null;
  sortOrder?: number;
  status?: number;
  description?: string;
}

// 产品列表响应
export interface ProductListResponse {
  id: number;
  name: string;
  categoryName: string;
  categoryId: number;
  manufacturer: string;
  model: string | null;
  price: string | null;
  summary: string | null;
  description: string;
  status: ProductStatus;
  views: number;
  images: string | null;
  featured: boolean;
}

// 产品详情响应
export interface ProductResponse {
  id: number;
  name: string;
  category: ProductCategoryResponse;
  manufacturer: string;
  model: string | null;
  price: string | null;
  summary: string | null;
  description: string;
  application: string;
  contactPhone: string;
  contactEmail: string;
  contact: string | null;
  website: string | null;
  status: ProductStatus;
  views: number;
  featured: boolean;
  createdTime: string;
  updatedTime: string;
  features: string | null;
  certifications: string | null;
  images: string | null;
  specifications: string | null;
}

// 创建/更新请求
export interface ProductRequest {
  name: string;
  categoryId?: number;
  manufacturer?: string;
  model?: string;
  price?: string;
  summary?: string;
  description?: string;
  application?: string;
  contactPhone?: string;
  contactEmail?: string;
  contact?: string;
  website?: string;
  status?: number;
  featured?: boolean;
  features?: string;
  certifications?: string;
  images?: string;
  specifications?: string;
}
